CMakeLists.txt

```diff
@@ -4,6 +4,11 @@ enable_testing()
 
 set(CMAKE_CXX_FLAGS "-std=c++11 -Wall -Wextra")
```

Comment:
While we're modifying this file, could we get rid of this and switch to the target based approach?

Reply:
Absolutely. I also made unit tests optional, they can be disabled on command line or by a superproject